src/components/Question.tsx

```diff
@@ -0,0 +1,166 @@
+/** @jsx jsx */
+import { jsx, css } from "@emotion/react";
+import React, { ChangeEvent } from "react";
+
+import { Question, QuestionType } from "../api/question";
+import { selectable } from "../commonStyles";
+import create_input from "./InputTypes";
+
+const _skip_normal_state: Array<QuestionType> = [
+    QuestionType.Radio,
+    QuestionType.Checkbox,
+    QuestionType.Select,
+    QuestionType.Section,
+    QuestionType.Range
+];
+
+export type QuestionProp = {
+    question: Question,
+    public_state: Map<string, string | boolean | null>,
+}
+
+class RenderedQuestion extends React.Component<QuestionProp> {
+    constructor(props: QuestionProp) {
+        super(props);
+        if (props.question.type === QuestionType.TextArea) {
+            this.handler = this.text_area_handler.bind(this);
+        } else {
+            this.handler = this.normal_handler.bind(this);
+        }
+
+        if (!_skip_normal_state.includes(props.question.type)) {
+            this._setState("value", "");
+        }
+    }
+
+    _setState(target: string, value: string | boolean | null, callback?:() => void): void {
```

Comment:
Nicer method name, underscores aren't so nice:
```suggestion
            this.setPublicState("value", "");
        }
    }

    setPublicState(target: string, value: string | boolean | null, callback?:() => void): void {
```